packages/grafana-data/src/types/datasource.ts

```diff
@@ -351,6 +351,19 @@ abstract class DataSourceApi<
    * @alpha -- experimental
    */
   getDefaultQuery?(app: CoreApp): Partial<TQuery>;
+
+  /*
+   * Optionally, use this method to track information about queries.
+   * Only executed once, during the dashboard initialization.
+   * @alpha -- experimental
+   */
+  trackQueries?(
+    queries: TQuery[],
+    dashboardId?: string,
+    orgId?: number,
+    userId?: number,
+    grafanaVersion?: string
```

Comment:
**Suggestion:** I think we should rather use a single object argument here instead of positional arguments

My reasoning is that if plugins start implementing this method and after some time we decide to remove some information from what we pass down, then we can introduce breaking changes if we use  positional parameters. 

As an example let's say that we decide to remove the `orgId` from what we pass down. In this case the dependant plugins need to update their code, otherwise they will track the `userId` as `orgId`. On the other hand if we use an object (with mostly optional fields), then in worst case they will start tracking `undefined` values for the removed fields, but they won't mix up.

```javascript
trackQueries?(options: {
    queries: TQuery[],
    dashboardId?: string,
    orgId?: number,
    userId?: number,
    grafanaVersion?: string
});
```

Reply:
Agreed, I think that makes sense, I'll update 👍